### ISSUE TITLE
Release v0.1.0

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-dotnet-instrumentation#471]
1. Remember to run tests on .NET 4.6.2 locally (they do not run in GitHub)
2. Follow `docs/releasing.md`

**Comment by Kielek** [comment]
Please check the versioning issue on OTel SDK side. We should use `-beta.1` suffix instead ot `-beta1`

https://github.com/open-telemetry/opentelemetry-dotnet/issues/2991#issuecomment-1083242602

**Comment by pellared** [comment]
**SIG meeting**:

1. Create a blog post on https://opentelemetry.io/blog/ (probably PR here: https://github.com/open-telemetry/opentelemetry.io_) - TODO create GH issue
2. Announcement on OTel .NET SDK Slack
3. Announcement during OTel .NET SIG meeting 
4. Ask customers to give it a try
5. @noahfalk can this be advertised via [official .NET Blog](https://devblogs.microsoft.com/dotnet/), [dotNet YouTube channel](https://www.youtube.com/channel/UCvtT19MZW8dq5Wwfu6B0oxw), [Microsoft Build](https://mybuild.microsoft.com/en-US/home), etc?
6. Ask in README.md for experience reports via Slack or GH issues.
7. @nrcventura will try to get info regarding CNCF resources

**Comment by pellared** [comment]
> Please check the versioning issue on OTel SDK side. We should use `-beta.1` suffix instead ot `-beta1`
> 
> [open-telemetry/opentelemetry-dotnet#2991 (comment)](https://github.com/open-telemetry/opentelemetry-dotnet/issues/2991#issuecomment-1083242602)

I was considering simply making a `v0.1.0` tag and name the release `0.1.0`. I think that tags like `-beta.1` or `-rc.1` are more important for `1.x.y` versions. For `0.x.y` I do not find it important.

@open-telemetry/dotnet-maintainers do you agree?

**Comment by pjanotti** [comment]
IIRC semantic version supports the suffix. Given that I have to ask: does it makes anything easier if you drop it? If yes, I'm ok with dropping it, otherwise, why not use it?

**Comment by pjanotti** [comment]
> v0.1.0-beta.1 for the git tag? How do we want to name the release? 0.1.0 beta1?

Yes, for the tag. For text and communication the `beta1` seems fine for me. 

We should increase the patch number as recommended for semver, but, unless we add some significative change or feature it should be our "beta one".

**Comment by nrcventura** [comment]
> We should increase the patch number as recommended for semver, but, unless we add some significative change or feature it should still be our "beta one".
> 
> I mean we may have a "beta two" so I don't see us changing the "beta.1" suffix soon.

Just to clarify my understanding. When we ship another prerelease that includes MongoDB support, which version will that be?

1.  `0.1.1`
2. `0.1.1-beta.1`
3. `0.1.0-beta.2`
4. `0.2.0`
5. `0.2.0-beta.1`
6. something else?

**Comment by pjanotti** [comment]
My preference is for option 2: `0.1.1-beta.1`

We added one bytecode instrumentation but the general features of automatic instrumentation are the same. Suppose that after some feedback and bugs we change some defaults, modify how bytecode instrumentation is added, and so on. At that point we name it `0.2.0-beta.2`

**Comment by pellared** [comment]
Then why not simply 0.1.1 and 0.2.0? the minor number seems to be equal to the beta number. Any case when you want to have it different?

How we want to version when we bump OTel SDK?

**Comment by nrcventura** [comment]
> Is there any case when we want to have them different?

I don't have a good answer this. If everything less than 1.0.0 is considered a beta does the `-beta.x` provide any value? If we have `0.1.0-beta.1`, `0.1.1-beta.1`, and `0.2.0-beta.2` the `-beta.x` component becomes confusing to me. I usually think of the `x` in `-beta.x` as meaning this is the "xth" prerelease of the version number that preceded `-beta.x`.

**Comment by nrcventura** [comment]
Perhaps we should treat `0.1.0` as a "stable" preview version, whereas we treat `0.1.0-beta.x` as the `xth` beta release of `0.1.0`. Then as we make changes such as updating dependencies, adding metrics or logs support, or bug fixes, we have a conversation about whether the scope of the change is enough to bump `0.1.0-beta.1` to `0.2.0-beta.1` or `0.1.1-beta.1`, or if we should simply bump the beta number for these changes by changing `0.1.0-beta.1` to `0.1.0-beta2`.

The idea is to treat the versions < 1.0.0 the same as versions >= 1.0.0, with the main difference being that versions < 1.0.0 may contain breaking changes between minor and patch releases.

**Comment by noahfalk** [comment]
> @noahfalk can this be advertised via [official .NET Blog](https://devblogs.microsoft.com/dotnet/), [dotNet YouTube channel](https://www.youtube.com/channel/UCvtT19MZW8dq5Wwfu6B0oxw), [Microsoft Build](https://mybuild.microsoft.com/en-US/home), etc?

We can certainly reach out to the folks who decide content for each of those channels and suggest it, then they can make a call. Do you have any suggested text or a bullet point list of key messages you want to convey?

**Comment by pellared** [comment]
I am going to make a release named `0.1.0-beta.1` tagged `v0.1.0-beta.1`.

I think we should document our pre-release versioning approach in `VERSIONING.md`  before making the next release (which I hope we will do rather sooner than later 😉 Let's continue the discussions here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/474

**Comment by pellared** [comment]
> > @noahfalk can this be advertised via [official .NET Blog](https://devblogs.microsoft.com/dotnet/), [dotNet YouTube channel](https://www.youtube.com/channel/UCvtT19MZW8dq5Wwfu6B0oxw), [Microsoft Build](https://mybuild.microsoft.com/en-US/home), etc?
> 
> Do you have any suggested text or a bullet point list of key messages you want to convey?

@noahfalk 
I think we can propose a text once we have the release done and a blog entry on https://opentelemetry.io/blog/ published (tracked under https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/572). We will let you know when we are ready.

**Comment by noahfalk** [comment]
To set expectations whether the .NET team talks about it on official channels depends on what is being released. Nothing in this issue says what the thing being released is : )

**Comment by pellared** [comment]
@noahfalk beta version of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation is going to be released (working in it). Here is what will be included in the release (currently as Unreleased): https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/CHANGELOG.md

**Comment by Kielek** [comment]
Beta release - https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/v0.1.0-beta.1
